### PR TITLE
Updating default.tensor.tf

### PR DESCRIPTION
**Pull request by co9olguy**
This PR brings the `default.tensor.tf` in line with recent upgrades to the parent device`defaul.tensor`.

To do:

- [ ] Update `default.tensor.tf` to get `test_default_tensor_tf.py` tests passing
- [x] Move core implementation of operations and expectations into separate `tf_ops.py` file
- [x] Update `default.tensor.tf` to take into account new features of `default.tensor`
- [x] Update any relevant docstrings

Nice to haves:
- [ ] Make sure `default.tensor.tf` can pass all the tests in `test_default_tensor.py`?

Notes:

TensorFlow v2.1+ is needed in order to support gradients of complex-valued svd (needed for `mps` representation). We will have to determine if there are issues along the lines of those mentioned here: https://github.com/tensorflow/tensorflow/issues/37307

### CODE REVIEW

**Comment by co9olguy** [comment]
@josh146 I managed to port everything over and get all existing tests to pass, except for one: `test_jacobian_agrees` now fails when using the "finite-diff" grad method. I am unable to figure out why this test fails when it passed before

File: pennylane/beta/plugins/default_tensor_tf.py [hunk 1]
```diff
@@ -252,11 +116,13 @@ class DefaultTensorTF(DefaultTensor):
...
+    _squeeze = staticmethod(tf.squeeze)
+    _expand_dims = staticmethod(tf.expand_dims)
 
-    C_DTYPE = C_DTYPE
-    R_DTYPE = R_DTYPE
+    C_DTYPE = ops.C_DTYPE
+    R_DTYPE = ops.R_DTYPE
 
-    def __init__(self, wires, shots=1000):
+    def __init__(self, wires, shots=1000, representation="exact", contraction_method="auto"):
```
Comment (josh146) [inline pennylane/beta/plugins/default_tensor_tf.py:125]: Sweet!

File: pennylane/beta/plugins/default_tensor_tf.py [hunk 2]
```diff
@@ -373,6 +244,7 @@ def jacobian(self, queue, observables, parameters):
         Returns:
             array[float]: Jacobian matrix of size (``num_params``, ``num_wires``)
         """
+        self.reset()
```
Comment (josh146) [inline pennylane/beta/plugins/default_tensor_tf.py:247]: The changes required here were much smaller than I expected 🤯

File: pennylane/beta/plugins/tf_ops.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
```
Comment (josh146) [inline pennylane/beta/plugins/tf_ops.py:1]: Nice having this as a separate module!

File: tests/beta/test_default_tensor_tf.py [hunk 4]
```diff
@@ -97,74 +102,77 @@ def _init_state(n):
 #####################################################
 
 
+@pytest.mark.parametrize("rep", ("exact", "mps"))
```
Comment (josh146) [inline tests/beta/test_default_tensor_tf.py:105]: :+1:

File: tests/beta/test_default_tensor_tf.py [hunk 5]
```diff
@@ -97,74 +102,77 @@ def _init_state(n):
...
+    def test_invalid_qubit_state_vector(self, rep):
         """Test that an exception is raised if the state
         vector is the wrong size"""
-        dev = DefaultTensorTF(wires=2)
+        dev = DefaultTensorTF(wires=2, representation=rep)
         state = np.array([0, 123.432])
 
-        with pytest.raises(ValueError, match=r"State vector must be of length 2\*\*wires"):
+        with pytest.raises(
+            ValueError, match=r"can apply QubitStateVector only to all of the 2 wires"
```
Comment (josh146) [inline tests/beta/test_default_tensor_tf.py:140]: Just noticing this now, but this sounds a bit weird. "can _only_ apply QubitStateVector to all 2 wires" sounds better

**Comment by josh146** [comment]
A couple of notes on the test failures:

* On Python 3.5, it looks like TensorNetwork is not supported. Although, I had a feeling that we took this into account?

* Could the failing jacobian test be because of the einsum issue? That's the only thing that makes sense to me, since the result of the simulator is correct, just not the Jacobian. My guess could be that the upgrades to `default.tensor` are resulting in `tensornetwork` using `einsum` behind the scenese where it previously wasn't?

  Although, correct me if I'm wrong, but if that was the case, I would expect the `exact` test to pass, but the `mps` test to fail?

**Comment by co9olguy** [comment]
>     * Could the failing jacobian test be because of the einsum issue? That's the only thing that makes sense to me, since the result of the simulator is correct, just not the Jacobian. My guess could be that the upgrades to `default.tensor` are resulting in `tensornetwork` using `einsum` behind the scenese where it previously wasn't?
>       Although, correct me if I'm wrong, but if that was the case, I would expect the `exact` test to pass, but the `mps` test to fail?

I believe I saw this test failing even with TF2.0

File: tests/beta/test_default_tensor_tf.py (hunk 5 above)
Comment (co9olguy) [reply tests/beta/test_default_tensor_tf.py:140]: I'll have to disagree here, I think it's fine as is

**Comment by co9olguy** [comment]
Fixed failing tests. Was due to that test comparing outputs between default.tensor.tf and default.qubit, but the two default step-sizes for finite-difference were not the same

**Pull request merged**